### ISSUE TITLE
[1822] - Companies can buy trains not the cheapest when Emergency buying with 2P trains

### ISSUE THREAD

**Issue by xasil** [tobymao/18xx#5532]
https://18xx.games/game/44254

Companies can buy any train from market , ignoring the must buy cheapest when Emergency buying when they have a 2P. Several E trains bought this way in the game included.

**Comment by jenf** [comment]
It's hard to find which action you mean, could you provide an action (use history mode to find it)

**Comment by KevinR1000** [comment]
Based on description that this involved the purchase of E trains, there are five relevant actions:

- LBSCR with 2P and £1123 buys E for £1000 -- https://18xx.games/game/44254?action=1072
- LYR with 2P and £1022 buys E for £1000 -- https://18xx.games/game/44254?action=1098
- NER with 7T and £151 buys E (from LYR) for £1 -- https://18xx.games/game/44254?action=1120
- MR with 5T and £1099 buys E for £1000 -- https://18xx.games/game/44254?action=1131
- SWR with 6T and £73 buys E (from MR) for £73 -- https://18xx.games/game/44254?action=1178

None of the purchases from the bank involved emergency money raising, so there was no requirement to buy the cheapest train.  Relevant rules are 5.14.2 (which applies to LBSCR and LYR) and 5.14.3 (emergency money raising, which applies to none of the companies) [emphasis added]:

> 5.14.2 
If a company that does not own a train **_has enough money_** to buy a train from the bank pool or bank, then it may do so. Alternatively, it may buy a train from another company, as described in section 5.13.3.
5.14.3 
If such a company **_cannot afford_** to buy a train from the bank pool or bank, it must buy either the **_cheapest_** train from the bank (including any trains in the bank pool) with help from the company’s director, or if the director wishes and can arrange a purchase, a train from another company. For the double-sided trains (L/2 and 7/E) this means that the cheaper version must always be the version bought if the purchase is made from the bank.

In short, all of the E train purchases appear fine to me.

**Comment by tobymao** [comment]
thanks for looking @KevinR1000
